uaclient/tests/test_config.py

```diff
@@ -1155,14 +1153,21 @@ class TestFeatures:
             ),
         ),
     )
+    @mock.patch("uaclient.config.logging.warning")
     def test_features_are_a_property_of_uaconfig(
-        self, cfg_features, expected, warnings, caplog_text, FakeConfig
+        self,
+        m_log_warning,
+        cfg_features,
+        expected,
+        warnings,
+        FakeConfig,
     ):
         user_cfg = {"features": cfg_features}
         cfg = FakeConfig(cfg_overrides=user_cfg)
         assert expected == cfg.features
         if warnings:
-            assert warnings in caplog_text()
+            expected = [mock.call(warnings, cfg_features)]
+            assert expected == m_log_warning.call_args_list
```

Comment:
It doesn't have any adverse affects here, but let's keep good practices. So don't overwrite the `expected` argument. Also no need to have the intermediate variable anyway
```suggestion
            assert [mock.call(warnings, cfg_features)] == m_log_warning.call_args_list
```

Reply:
Updated